app/onboarding/page.tsx

```diff
@@ -0,0 +1,170 @@
+  selectedGardenType: string;
+  setSelectedGardenType: React.Dispatch<React.SetStateAction<string>>;
+}
+
+interface PlotSelectionProps {
+  selectedPlot: boolean | null;
+  setSelectedPlot: React.Dispatch<React.SetStateAction<boolean>>;
+}
+
+// Select State
+const StateSelection: React.FC<StateSelectionProps> = ({
+  selectedState,
+  setSelectedState,
+}) => {
+  return (
+    <div>
+      <h2>Which state do you live in?</h2>
+      <select
+        value={selectedState}
+        onChange={e => setSelectedState(e.target.value)}
+      >
+        <option value="" disabled>
+          Select a state
+        </option>
+        {states.map(state => (
+          <option key={state} value={state}>
+            {state}
+          </option>
+        ))}
+      </select>
+    </div>
+  );
+};
+
+// Step 2: Select garden type
+
+const GardenTypeSelection: React.FC<GardenTypeSelectionProps> = ({
+  selectedGardenType,
+  setSelectedGardenType,
+}) => {
+  return (
+    <div>
+      <h2>What type of garden do you want to create?</h2>
+      {gardenTypes.map(type => (
+        <label key={type}>
+          <input
+            type="radio"
+            name="gardenType"
+            value={type}
+            checked={selectedGardenType === type}
+            onChange={e => setSelectedGardenType(e.target.value)}
+          />
+          {type}
+        </label>
+      ))}
+    </div>
+  );
+};
+
+// Step 3: Do you have a plot?
+const PlotSelection: React.FC<PlotSelectionProps> = ({
+  selectedPlot,
+  setSelectedPlot,
+}) => {
+  return (
+    <div>
+      <h2>Do you already have a plot?</h2>
+      {plotOptions.map(option => (
+        <label key={option.label}>
+          <input
+            type="radio"
+            name="plot"
+            value={String(option.value)}
+            checked={selectedPlot === option.value}
+            onChange={() => setSelectedPlot(option.value)}
+          />
+          {option.label}
+        </label>
+      ))}
+    </div>
+  );
+};
+
+// Main Onboarding Component
+const OnboardingFlow = () => {
+  const [step, setStep] = useState(1);
+  const [selectedState, setSelectedState] = useState<string>('');
+  const [selectedGardenType, setSelectedGardenType] = useState<string>('');
+  const [selectedPlot, setSelectedPlot] = useState<boolean>(false);
+
+  const handleNext = () => {
+    setStep(step + 1);
+  };
+
+  const handleBack = () => {
+    setStep(step - 1);
+  };
+  const { setProfile } = useProfile();
+  const handleSubmit = async () => {
+    try {
+      await setProfile({
+        state: selectedState,
+        user_type: selectedGardenType,
+        has_plot: selectedPlot,
+      });
+      console.log('Profile updated successfully');
+    } catch (error) {
+      console.error('Error submitting profile:', error);
+    }
+  };
+  // Handle form submission, e.g., send to a server or display a confirmation
+
+  return (
+    <ProfileProvider>
+      <div>
+        {step === 1 && (
```

Comment:
@ccatherinetan just flagging that on ijp, [each step of the onboarding flow was a separate page](https://github.com/calblueprint/immigration-justice-project/tree/main/src/app/onboarding). however in this case i think this is _probably_ fine because all the screens in the flow are so similar to each other, and this also prevents the need for an `OnboardingContext`!

Reply:
yup! ronnie also mentioned that he thinks multi-page onboarding is cleaner, but we can leave that refactor for next sem / as a stretch